tests/main/snap-quota-cpu/task.yaml

```diff
@@ -0,0 +1,129 @@
+summary: Functional test for cpu quota-related snap commands.
+
+details: |
+  Functional test for snap cpu quota group commands ensuring that they are 
+  effective in practice.
+
+# these systems do not support quota groups due to their old systemd versions,
+# we do have a check to do for these systems, but that is just to ensure that
+# we don't allow using quota groups, and that check is done in the snap-quota
+# spread instead
+systems:
+  - -ubuntu-14.04-*
+  - -centos-7-*
+  - -amazon-linux-2-*
+  - -ubuntu-16.04-*
+  - -ubuntu-core-16-*
+
+prepare: |
+  snap install test-snapd-stressd --edge --devmode
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  echo "Starting service and verifying cpu usage"
+  snap start test-snapd-stressd.stress-sc
+
+  # helper function that has retry logic to allow for cpu usage to stabilize
+  function ensure_cpu_usage_higher_than {
+    n=0
+    until [ "$n" -ge 5 ]
+    do
+      usage=$(top -b -n 2 -d 0.2 | head | grep "$1" | awk '{print $9*100}')
```

Comment:
Why do we need the `head` here? is it expensive to scan the entire thing? also `grep "$1"` will also find "stressme", "stressful" etc. Maybe use `grep " ${1}$"` here  ? also iirc/afaik top will use a limited line length, so maybe add `-w512` to increase the "width" ?

Reply:
Ok so I don't know if its okay to do - but the reason I 'head' this is to only get the top first few rows of the 'top' command, as the output from top is sorted by CPU usage (afaict), and in our test stress will always (atleast observed) be on top, so we don't need the entire output from top (which is *a lot* because of -n 2), and the reason for the grep was an assumption made that the only process in the first few lines would be stress for this test, I assumed the odds for another stress process to run in the test environment not started by the test was very close to 0. 

I'll happily change any of it as I don't have any strong feelings towards this, but I'll happily atleast add a comment for the assumptions made in that command line

Reply:
It's totally fine, I was just trying to understand the reasoning behind it :) All of what you write makes sense, I was reading with this the very critical "find corner-cases" code-review eye, hence the questions/suggestions. But of course in practice this  is fine and will work just fine (like you said, unlikely to have a different "stress-something" process :) Maybe a very small comment like: `# fine to use head here as top sorts by CPU usage`. I would suggest to use `top -w256` or something though, maybe I'm overly cautious but I feel uneasy that top by default will cutof the program line